app/vmui/packages/vmui/src/pages/ExploreLogs/ExploreLogs.tsx

```diff
@@ -8,45 +8,57 @@ import Spinner from "../../components/Main/Spinner/Spinner";
 import Alert from "../../components/Main/Alert/Alert";
 import ExploreLogsHeader from "./ExploreLogsHeader/ExploreLogsHeader";
 import "./style.scss";
-import { ErrorTypes } from "../../types";
+import { ErrorTypes, TimeParams } from "../../types";
 import { useState } from "react";
 import { useTimeState } from "../../state/time/TimeStateContext";
 import { getFromStorage, saveToStorage } from "../../utils/storage";
 import ExploreLogsBarChart from "./ExploreLogsBarChart/ExploreLogsBarChart";
 import { useFetchLogHits } from "./hooks/useFetchLogHits";
 import { LOGS_ENTRIES_LIMIT } from "../../constants/logs";
+import { getTimeperiodForDuration, relativeTimeOptions } from "../../utils/time";
 
 const storageLimit = Number(getFromStorage("LOGS_LIMIT"));
 const defaultLimit = isNaN(storageLimit) ? LOGS_ENTRIES_LIMIT : storageLimit;
 
 const ExploreLogs: FC = () => {
   const { serverUrl } = useAppState();
-  const { duration, relativeTime, period } = useTimeState();
+  const { duration, relativeTime, period: periodState } = useTimeState();
   const { setSearchParamsFromKeys } = useSearchParamsFromObject();
 
   const [limit, setLimit] = useStateSearchParams(defaultLimit, "limit");
   const [query, setQuery] = useStateSearchParams("*", "query");
+  const [period, setPeriod] = useState<TimeParams>(periodState);
   const { logs, isLoading, error, fetchLogs } = useFetchLogs(serverUrl, query, limit);
   const { fetchLogHits, ...dataLogHits } = useFetchLogHits(serverUrl, query);
   const [queryError, setQueryError] = useState<ErrorTypes | string>("");
   const [loaded, isLoaded] = useState(false);
   const [markdownParsing, setMarkdownParsing] = useState(getFromStorage("LOGS_MARKDOWN") === "true");
 
+  const getPeriod = useCallback(() => {
+    const relativeTimeOpts = relativeTimeOptions.find(d => d.id === relativeTime);
+    if (!relativeTimeOpts) return periodState;
+    const { duration, until } = relativeTimeOpts;
+    return getTimeperiodForDuration(duration, until());
+  }, [periodState, relativeTime]);
+
   const handleRunQuery = () => {
     if (!query) {
       setQueryError(ErrorTypes.validQuery);
       return;
     }
 
-    fetchLogs().then(() => {
+    const newPeriod = getPeriod();
+    setPeriod(newPeriod);
+    fetchLogs(newPeriod).then(() => {
+      fetchLogHits(newPeriod);
+    }).catch(e => e).finally(() => {
       isLoaded(true);
```

Comment:
i think this should  be false on finally, because we finished all requests. 

Maybe we can use `Promise.all` for two different request if they are independent. I mean for `fetchLogs` and `fetchLogHits`.

Reply:
1) The `isLoaded` flag was initially used to prevent displaying the "No logs found" alert before the request completed. Now, since we have a default expr, this flag is redundant, so I have removed it.
2) Executing the requests in parallel will reduce the loading time for the chart but will increase the load on the server.

Reply:
I think the server can handle it. But if you prefer to use this way it is ok.